### DISCUSSION TITLE
Adding subtitles after the recording

### DISCUSSION THREAD

**Discussion by bkil** [asciinema/discussions#91] [Ideas]
Please provide a way through which the author could add timed subtitles to a recording. This would usually be done after the session, but I could envision it being done in reverse: writing a subtitle to guide the presentation that gets shown live during recording as well.

Some users may choose to read subtitles via text to speech as well.

**Comment by bkil** [comment]
Related: https://github.com/asciinema/asciinema-server/issues/128

**Comment by arialdomartini** [comment]
A possible approach is to run the session in GNU screen configured to have a blank line in the bottom of the screen, to be filled with subtitles.

This has been automated by @hydrargyrum in [asciinema-comment](https://github.com/hydrargyrum/asciinema-comment).

**Comment by bkil** [comment]
Sorry, but that workaround is not acceptable. That's equivalent to "burning in" subtitles within videos. It would practically be unchangeable after the recording is made (i.e., to correct typos). You can't plausibly add subtitles in multiple languages either. It couldn't be processed ergonomically via screen readers either.

**Comment by arialdomartini** [comment]
@bkil are you seriously downvoting for sharing the info I had on a possible approach? Funny way to collaborate.

**Comment by hydrargyrum** [comment]
thanks for the reference @arialdomartini :) !

fyi i also made a program (still based upon asciinema) with another approach of scripting the full video: subtitles + typed stuff. This is [asciinario](https://gitlab.com/hydrargyrum/asciinario) (i should mirror it to github in the next days to make it more visible).

@bkil given the subtitles are added in post-processing (original cast + subtitles -> final cast), you can still re-generate a new video from the original cast that has no subtitles :) Also, as they're in reserved area of the screen (pun not intended), i didn't think about it much, but i guess you could replace them without mangling the rest of the cast even if you lost the source cast.
